src/Handler/SelfUpdate.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+/*
+ * This file is part of the Manalize project.
+ *
+ * (c) Manala <contact@manala.io>
+ *
+ * For the full copyright and license information, please refer to the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Manala\Manalize\Handler;
+
+use GuzzleHttp\Client;
+use Manala\Manalize\Application;
+use Manala\Manalize\Exception\HandlingFailureException;
+
+/**
+ * @author Robin Chalas <robin.chalas@gmail.com>
+ */
+class SelfUpdate implements Handler
```

Comment:
I wonder if we really need this interface. Actually we're never using it but instead clearly relies on implementations (which is not something bad here).